framework/src/userobject/ElementLoopUserObject.C

```diff
@@ -0,0 +1,147 @@
+#include "ElementLoopUserObject.h"
+
+template<>
+InputParameters validParams<ElementLoopUserObject>()
+{
+  InputParameters params = validParams<GeneralUserObject>();
+  return params;
+}
+
+ElementLoopUserObject::ElementLoopUserObject(const InputParameters & parameters) :
+    GeneralUserObject(parameters),
+    ThreadedElementLoop<ConstElemRange>(*getParam<FEProblem *>("_fe_problem")),
+    Coupleable(this, false),
+    MooseVariableDependencyInterface(),
+    ZeroInterface(parameters),
+    _mesh(_subproblem.mesh()),
+    _current_elem(_assembly.elem()),
+    _current_elem_volume(_assembly.elemVolume()),
+    _q_point(_assembly.qPoints()),
+    _qrule(_assembly.qRule()),
+    _JxW(_assembly.JxW()),
+    _coord(_assembly.coordTransformation()),
+    _have_interface_elems(false)
+{
+  // Keep track of which variables are coupled so we know what we depend on
+  const std::vector<MooseVariable *> & coupled_vars = getCoupledMooseVars();
+  for (unsigned int i = 0; i < coupled_vars.size(); i++)
+    addMooseVariableDependency(coupled_vars[i]);
+}
+
+ElementLoopUserObject::ElementLoopUserObject(ElementLoopUserObject & x, Threads::split split) :
+    GeneralUserObject(x.parameters()),
+    ThreadedElementLoop<ConstElemRange>(x.ThreadedElementLoop<ConstElemRange>::_fe_problem),
+    Coupleable(this, false),
+    MooseVariableDependencyInterface(),
+    ZeroInterface(x.parameters()),
+    _mesh(x._subproblem.mesh()),
+    _current_elem(x._assembly.elem()),
+    _current_elem_volume(x._assembly.elemVolume()),
+    _q_point(x._assembly.qPoints()),
+    _qrule(x._assembly.qRule()),
+    _JxW(x._assembly.JxW()),
+    _coord(x._assembly.coordTransformation()),
+    _have_interface_elems(false)
+{
+  // Keep track of which variables are coupled so we know what we depend on
+  const std::vector<MooseVariable *> & coupled_vars = x.getCoupledMooseVars();
+  for (unsigned int i = 0; i < coupled_vars.size(); i++)
+    addMooseVariableDependency(coupled_vars[i]);
+}
+
+ElementLoopUserObject::~ElementLoopUserObject()
+{
+}
```

Comment:
empty destructors are superflouous (if the base class already declared them virtual)